### PR TITLE
FIX: Throw value error for unsupported version of Mechanical

### LINKED ISSUE #913: Throw error if unsupported version of Mechanical is used with latest version of PyMechanical
Currently PyMechanical supports Mechanical from 23R2 and later versions.

### PR DESCRIPTION
**Pull request by dipinknair**
Fixed #913 

if no version is provided to ``App()`` takes the latest Mechanical version found in system and throws warning. 
In case of Linux, ``mechanical-env`` ensures only one version (either from saved path or latest version found in default installations directory) is found when checking AWP_ROOT env.

### CODE REVIEW

File: src/ansys/mechanical/core/embedding/initializer.py [hunk 1]
```diff
@@ -154,6 +154,8 @@ def initialize(version: int = None):
 
     if version == None:
         version = _get_default_version()
+    elif version not in SUPPORTED_MECHANICAL_EMBEDDING_VERSIONS_WINDOWS:
```
Comment (koubaa) [inline src/ansys/mechanical/core/embedding/initializer.py:157]: what about 251?

File: src/ansys/mechanical/core/embedding/initializer.py [hunk 2]
```diff
@@ -142,6 +142,16 @@ def __check_loaded_libs(version: int = None):  # pragma: no cover
         )
 
 
+def __check_for_supported_version(version):
+    # if below env is set, then users can overwrite version support
+    is_old_version = os.getenv("ANSYS_MECHANICAL_EMBEDDING_SUPPORT_OLD_VERSIONS")
+
+    if is_old_version is not None and is_old_version == "1" and version >= 231:
+        return version
+    elif version < next(iter(SUPPORTED_MECHANICAL_EMBEDDING_VERSIONS_WINDOWS)):
```
Comment (koubaa) [inline src/ansys/mechanical/core/embedding/initializer.py:151]: minor - "WINDOWS" in this variable name is unnecessary

**Review by koubaa — CHANGES_REQUESTED** [review]

File: src/ansys/mechanical/core/embedding/initializer.py [hunk 3]
```diff
@@ -142,6 +142,16 @@ def __check_loaded_libs(version: int = None):  # pragma: no cover
         )
 
 
+def __check_for_supported_version(version):
+    # if below env is set, then users can overwrite version support
+    is_old_version = os.getenv("ANSYS_MECHANICAL_EMBEDDING_SUPPORT_OLD_VERSIONS")
+
+    if is_old_version is not None and is_old_version == "1" and version >= 231:
```
[the remark this answers is not in the document]
Comment (dipinknair) [reply src/ansys/mechanical/core/embedding/initializer.py:149]: there is no support for version less than 231 right ?

File: src/ansys/mechanical/core/embedding/resolver.py [hunk 4]
```diff
@@ -37,5 +37,8 @@ def resolve(version):
     import Ansys  # isort: skip
 
     assembly_resolver = Ansys.Mechanical.Embedding.AssemblyResolver
-    resolve_handler = assembly_resolver.MechanicalResolveEventHandler
+    if version == 231:  # pragma: no cover
```
[the remark this answers is not in the document]
Comment (dipinknair) [reply src/ansys/mechanical/core/embedding/resolver.py:40]: Yes

[the remark this answers is not in the document]
Comment (dipinknair) [reply src/ansys/mechanical/core/embedding/resolver.py:40]: This will removed as per our discussion.

**Review by koubaa — APPROVED** [review]

**Pull request merged**